symplyphysics/core/vectors/arithmetics.py

```diff
@@ -38,37 +38,52 @@ def equal_vectors(vector_left: Vector, vector_right: Vector) -> bool:
-        )
-    if vector_left.coordinate_system.coord_system_type != CoordinateSystem.System.CARTESIAN:
-        coord_name_from = CoordinateSystem.system_to_transformation_name(
-            vector_left.coordinate_system.coord_system_type)
-        raise ValueError(
-            f"Addition is only supported for cartesian coordinates: got {coord_name_from}")
-    if vector_right.coordinate_system.coord_system_type != CoordinateSystem.System.CARTESIAN:
-        coord_name_from = CoordinateSystem.system_to_transformation_name(
-            vector_right.coordinate_system.coord_system_type)
-        raise ValueError(
-            f"Addition is only supported for cartesian coordinates: got {coord_name_from}")
-    (list_left_extended, list_right_extended) = _extend_two_vectors(vector_left, vector_right)
-    result = list(
-        map(lambda lr: sympify(lr[0] + lr[1]), zip(list_left_extended, list_right_extended)))
-    return Vector(result, vector_left.coordinate_system)
-
-
-def subtract_cartesian_vectors(vector_left: Vector, vector_right: Vector) -> Vector:
-    return add_cartesian_vectors(
-        vector_left,
-        scale_vector(-1, vector_right)
-    )
+def add_cartesian_vectors(*vectors: Vector) -> Vector:
+    def add_two_cartesian_vectors(vector_left: Vector, vector_right: Vector) -> Vector:
+        if vector_left.coordinate_system != vector_right.coordinate_system:
+            raise ValueError(
+                "Vectors must have the same coordinate system, "
+                f"got {vector_left.coordinate_system} and {vector_right.coordinate_system}"
+            )
+
+        left_type = vector_left.coordinate_system.coord_system_type
+        right_type = vector_right.coordinate_system.coord_system_type
+
+        if left_type != CoordinateSystem.System.CARTESIAN or right_type != CoordinateSystem.System.CARTESIAN:
+            left_name = CoordinateSystem.system_to_transformation_name(left_type)
+            right_name = CoordinateSystem.system_to_transformation_name(right_type)
+            raise ValueError(f"Expect two Cartesian vectors, got {left_name} and {right_name}")
+
+        (list_left_extended, list_right_extended) = _extend_two_vectors(vector_left, vector_right)
+        result = [
+            sympify(left_component + right_component)
+            for (left_component, right_component)
+            in zip(list_left_extended, list_right_extended)
+        ]
+
+        return Vector(result, vector_left.coordinate_system)
+
+    if not vectors:
+        return Vector([0, 0, 0])
+
+    result = vectors[0]
+    for vector in vectors[1:]:
+        result = add_two_cartesian_vectors(result, vector)
+    return result
+
+
+# Subtract vectors from the first given vector
+def subtract_cartesian_vectors(*vectors: Vector) -> Vector:
+    if not vectors:
+        raise ValueError("Provide at least one vector")
+
+    result = vectors[0]
+    for vector in vectors[1:]:
```

Comment:
I think easier solution is to add all vectors but first. And subtract sum from first vector.

Reply:
Agree